feat: validate attestationsHash and payloadDigest when synching checkpoints.

LHerskind opened the pull request:
Initial work to validate `attestationsHash` and `payloadDigest` when synching. Extends the event on L1, so needs a rollup upgrade to make a diff. Currently throwing if not matching, will need additional work to validate properly

LHerskind commented:
* **#19003** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19003?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19003?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

spalladino commented on yarn-project/archiver/src/archiver/l1/calldata_retriever.test.ts:
```diff
@@ -43,8 +52,13 @@ class TestCalldataRetriever extends CalldataRetriever {
+    proposeCalldata: Hex,
+    blockHash: Hex,
+    checkpointNumber: CheckpointNumber,
+    expectedHashes?: { attestationsHash?: Hex; payloadDigest?: Hex },
```
```suggestion
    expectedHashes: { attestationsHash?: Hex; payloadDigest?: Hex },
```
Nit, so we don't accidentally forget passing this argument in a callsite

spalladino commented on yarn-project/archiver/src/archiver/l1/calldata_retriever.ts:
```diff
@@ -365,6 +389,74 @@ export class CalldataRetriever {
+              signatureIndices: packedAttestations.signatureIndices,
+              signaturesOrAddresses: packedAttestations.signaturesOrAddresses,
+            },
+          ],
+        ),
+      );
```
Is this something we can retrieve from the abi instead of hardcoding here?

spalladino commented on yarn-project/archiver/src/archiver/l1/calldata_retriever.ts:
```diff
@@ -365,6 +389,74 @@ export class CalldataRetriever {
+        );
+      }
+
+      this.logger.trace(`LASSE: Validated attestationsHash for checkpoint ${checkpointNumber}`, {
```
```suggestion
      this.logger.trace(`Validated attestationsHash for checkpoint ${checkpointNumber}`, {
```

spalladino reviewed (approved):
Looks good! I'd just remove the `LASSE` from the logs :-P

LHerskind commented:
> Looks good! I'd just remove the `LASSE` from the logs :-P

Oh, I forgot 😂

spalladino commented on yarn-project/archiver/src/archiver/l1/calldata_retriever.ts:
```diff
@@ -324,17 +344,59 @@ export class CalldataRetriever {
+   * Finds the _attestations parameter by name in the propose function.
+   * Lazy-loaded to avoid issues during module initialization.
+   */
+  private getCommitteeAttestationsStructDef(): AbiParameter {
```
I'm surprised viem doesn't have a fancy typed function for extracting something from the abi. Oh well.

LHerskind committed:
feat: validate attestationsHash and payloadDigest when synching checkpoints.

Initial work to validate `attestationsHash` and `payloadDigest` when synching. Extends the event on L1, so needs a rollup upgrade to make a diff. Currently throwing if not matching, will need additional work to validate properly

spalladino commented:
Rebasing so we get Alex's fix for the e2e settings flake
